kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -129,7 +129,7 @@ private[kyo] class IOPromise[T](state: State[T])
             promise.get() match
                 case _: Pending[T] @unchecked =>
                     IOs {
-                        Scheduler.flush()
+                        Scheduler.get.flush()
```

Comment:
`Scheduler` isn't an object anymore. The global instance returned by `get` isn't lazily initialized because it'd require a memory barrier to read the field. It uses a regular `val` loaded when the companion object class is loaded.